src/cache.test.ts

```diff
@@ -0,0 +1,177 @@
+    expect(cache.get('d')).toBe(4)
+  })
+
+  test('should update existing key without increasing size', () => {
+    cache.set('a', 1)
+    cache.set('b', 2)
+    cache.set('a', 10) // Update existing key
+    expect(cache.size).toBe(2)
+    expect(cache.get('a')).toBe(10)
+  })
+
+  test('should check existence with has()', () => {
+    cache.set('a', 1)
+    expect(cache.has('a')).toBe(true)
+    expect(cache.has('b')).toBe(false)
+  })
+
+  test('should delete items', () => {
+    cache.set('a', 1)
+    expect(cache.delete('a')).toBe(true)
+    expect(cache.get('a')).toBeUndefined()
+    expect(cache.delete('nonexistent')).toBe(false)
+  })
+
+  test('should clear all items', () => {
+    cache.set('a', 1)
+    cache.set('b', 2)
+    cache.clear()
+    expect(cache.size).toBe(0)
+    expect(cache.get('a')).toBeUndefined()
+  })
+
+  test('should handle maxSize of 1', () => {
+    const smallCache = new LRUCache<string, number>(1)
+    smallCache.set('a', 1)
+    smallCache.set('b', 2)
+    expect(smallCache.size).toBe(1)
+    expect(smallCache.get('a')).toBeUndefined()
+    expect(smallCache.get('b')).toBe(2)
+  })
+
+  test('should expire entries after max-age', () => {
+    // Create cache with 100ms max-age for testing
+    const ttlCache = new LRUCache<string, number>(3, 100)
+    const originalNow = Date.now
+
+    try {
+      let currentTime = originalNow()
+      Date.now = () => currentTime
+
+      ttlCache.set('a', 1)
+      expect(ttlCache.get('a')).toBe(1)
+
+      // Advance time past expiration (101ms)
+      currentTime += 101
+      expect(ttlCache.get('a')).toBeUndefined()
+    } finally {
+      // Restore Date.now
+      Date.now = originalNow
+    }
+  })
+
+  test('should return false for has() on expired entries', () => {
+    const ttlCache = new LRUCache<string, number>(3, 100)
+    const originalNow = Date.now
+
+    try {
+      let currentTime = originalNow()
+      Date.now = () => currentTime
+
+      ttlCache.set('a', 1)
+      expect(ttlCache.has('a')).toBe(true)
+
+      // Advance time past expiration
+      currentTime += 101
+      expect(ttlCache.has('a')).toBe(false)
+    } finally {
+      // Restore Date.now
+      Date.now = originalNow
+    }
+  })
+
+  test('should use default max-age of 1 hour', () => {
+    const defaultCache = new LRUCache<string, number>(3)
+    const originalNow = Date.now
+
+    try {
+      let currentTime = originalNow()
+      Date.now = () => currentTime
+
+      defaultCache.set('a', 1)
+      expect(defaultCache.get('a')).toBe(1)
+
+      // Advance time just under 1 hour (59 minutes)
+      currentTime += 59 * 60 * 1000
+      expect(defaultCache.get('a')).toBe(1)
+
+      // Advance time past 1 hour total (2 more minutes)
+      currentTime += 2 * 60 * 1000
+      expect(defaultCache.get('a')).toBeUndefined()
+    } finally {
+      // Restore Date.now
+      Date.now = originalNow
+    }
+  })
+})
+
+describe('generateCacheKey', () => {
```

Comment:
Add a collision test making sure that generateCacheKey('abc', 'def', 'ghi') doesn't collide with ('ab', 'cde', 'fghi')